### PR TITLE
Add auto-complete support for JOIN and related keywords.

### PR DESCRIPTION
**Pull request by getaaron**
## Description

Adds auto-complete support for `JOIN` and [other join types](https://docs.aws.amazon.com/athena/latest/ug/select.html). Also adds `ON` and `USING`.

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: test/test_naive_completion.py [hunk 1]
```diff
@@ -52,3 +52,21 @@ def test_column_name_completion(completer, complete_event):
         Document(text=text, cursor_position=position),
         complete_event)
     assert result == list(map(Completion, sorted(completer.all_completions)))
+
+def test_various_join_completions(completer, complete_event):
+    for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
+        text = 'SELECT foo FROM bar {join_type} '
```
Comment (zzl0) [inline test/test_naive_completion.py]: What's the value of `'SELECT foo FROM bar {join_type} '`? should we use f-Strings here?

File: test/test_naive_completion.py [hunk 2]
```diff
@@ -52,3 +52,21 @@ def test_column_name_completion(completer, complete_event):
...
+        position = len(text)
+        result = completer.get_completions(
+            Document(text=text, cursor_position=position),
+            complete_event)
+        assert Completion(text='JOIN') in result
+
+def test_outer_join_completion(completer, complete_event):
+    for join_type in ['LEFT', 'RIGHT', 'FULL']:
+        text = 'SELECT foo FROM bar {join_type} '
```
Comment (zzl0) [inline test/test_naive_completion.py]: same here

File: test/test_naive_completion.py (hunk 1 above)
Comment (getaaron) [reply test/test_naive_completion.py]: I’m new to Python, so unsure about your comment. I looked up F strings but I don’t see the benefit here. I have the test in a loop so we don’t need a separate test for each keyword.

How would F strings help here?

Comment (zzl0) [reply test/test_naive_completion.py]: f-strings will replace the variable with its value, but the normal string will do nothing. e.g.:
``` python
>>> for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
...     print('SELECT foo FROM bar {join_type} ')
...
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}
SELECT foo FROM bar {join_type}

>>> for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
...     print(f'SELECT foo FROM bar {join_type} ')  # f-strings
...
SELECT foo FROM bar INNER
SELECT foo FROM bar OUTER
SELECT foo FROM bar CROSS
SELECT foo FROM bar LEFT
SELECT foo FROM bar RIGHT
SELECT foo FROM bar FULL
```

Comment (getaaron) [reply test/test_naive_completion.py]: @zzl0 I'm kind of surprised the tests were passing as-is, but I see the mistake now. Thanks for the explanation. I have pushed a fix.

**Comment by getaaron** [comment]
@zzl0 Looks like the F-strings don't work in the older versions of Python: https://travis-ci.org/dbcli/athenacli/builds/651010661

I pushed an update that uses string concatenation instead, let me know if this looks OK to you.

File: test/test_naive_completion.py [hunk 3]
```diff
@@ -52,3 +52,21 @@ def test_column_name_completion(completer, complete_event):
...
+        text = 'SELECT foo FROM bar ' + join_type + ' '
+        position = len(text)
+        result = completer.get_completions(
+            Document(text=text, cursor_position=position),
+            complete_event)
+        assert Completion(text='JOIN') in result
+
+def test_outer_join_completion(completer, complete_event):
+    for join_type in ['LEFT', 'RIGHT', 'FULL']:
+        text = 'SELECT foo FROM bar ' + join_type + ' '
```
Comment (zzl0) [inline test/test_naive_completion.py]: Yes, this is fine, thanks. BTW, we plan to stop supporting old versions of Python (e.g. Python2). After that, we will be able to use f-strings.

File: test/test_naive_completion.py (hunk 1 above)
Comment (zzl0) [reply test/test_naive_completion.py]: > I'm kind of surprised the tests were passing as-is

Could you try to set parameter `smart_completion=True` when calling `get_completions`? this should fail the first version of your tests.

File: test/test_naive_completion.py [hunk 4]
```diff
@@ -52,3 +52,21 @@ def test_column_name_completion(completer, complete_event):
...
         complete_event)
     assert result == list(map(Completion, sorted(completer.all_completions)))
+
+def test_various_join_completions(completer, complete_event):
+    for join_type in ['INNER', 'OUTER', 'CROSS', 'LEFT', 'RIGHT', 'FULL']:
+        text = 'SELECT foo FROM bar ' + join_type + ' '
+        position = len(text)
+        result = completer.get_completions(
+            Document(text=text, cursor_position=position),
+            complete_event)
```
[the remark this answers is not in the document]
Comment (getaaron) [reply test/test_naive_completion.py]: With `smart_completion=True` the tests I expected to fail did when I reverted back to the earlier string format. Thanks for your help with the tests. I think this is good to go now.

**Pull request merged**